automatminer/automl/gen_evolve_v2.py

```diff
@@ -0,0 +1,101 @@
+import random
+
+class NeuralNetOptimizer():
+    def __init__(self, X, y, model, retain = 5, random_select=0.05, mutate_chance=0.05, count=15):
+        self.random_select = random_select
+        self.mutate_chance = mutate_chance
+        self.retain = retain
+        self.X = X
+        self.y = y
+        self.Model = model
+        ACTIVATION = ["sigmoid", "tanh", "relu", "elu"]
+        OPTIMIZERS = ["sgd", "rmsprop", "adagrad", "adadelta", "nadam", "adamax", "adam"]
+        self.NN_PARAM_CHOICES = {"units": range(1, 1000), "hidden_layer_sizes": range(1, 3), "optimizer": OPTIMIZERS,
+                            "activation": ACTIVATION}
+        pop = self.create_population(count)
+        for _ in range(10):
+            pop = self.evolve(pop, X, y)
```

Comment:
Just initializing the class should probably not run any super expensive operations. We are just initializing the class, after all - this seems to be running the entire GA (including training etc.!!) right here in __init__.

This is the opposite problem we have with self.create_population. This probably should have its own method, e.g., `optimize`.